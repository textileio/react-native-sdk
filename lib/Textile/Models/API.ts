export interface DiscoveredCafe {
  readonly peer: string
  readonly address: string
  readonly api: string
  readonly protocol: string
  readonly node: string
  readonly url: string
}

export interface DiscoveredCafes {
  readonly primary: DiscoveredCafe
  readonly secondary: DiscoveredCafe
}

export interface File {
  readonly mill: string
  readonly checksum: string
  readonly source: string
  readonly opts?: string
  readonly hash: string
  readonly key?: string
  readonly media: string
  readonly name?: string
  readonly size: number
  readonly added: string
  readonly meta?: { readonly [key: string]: any }
  readonly targets?: ReadonlyArray<string>
}

export enum BlockType {
  MERGE = 'MERGE',
  IGNORE = 'IGNORE',
  FLAG = 'FLAG',
  JOIN = 'JOIN',
  ANNOUNCE = 'ANNOUNCE',
  LEAVE = 'LEAVE',
  MESSAGE = 'MESSAGE',
  FILES = 'FILES',
  COMMENT = 'COMMENT',
  LIKE = 'LIKE',
  INVALID = 'INVALID'
}

export interface BlockInfo {
  readonly id: string
  readonly thread_id: string
  readonly author_id: string
  readonly username: string
  readonly avatar?: string
  readonly type: BlockType
  readonly date: string
  readonly parents: ReadonlyArray<string>
  readonly target?: string
  readonly body?: string
}

export interface ExternalInvite {
  readonly id: string
  readonly key: string
  readonly inviter: string
}

export interface Cafe {
  readonly peer: string
  readonly address: string
  readonly api: string
  readonly protocol: string
  readonly node: string
  readonly url: string
  readonly swarm: ReadonlyArray<string>
}

export interface ContactInfo {
  readonly id: string
  readonly address: string
  readonly username?: string
  readonly avatar?: string
  readonly inboxes?: ReadonlyArray<Cafe>
  readonly created: string
  readonly updated: string
  readonly thread_ids?: ReadonlyArray<string>
}

export interface ContactInfoQueryResult {
  readonly local?: ReadonlyArray<ContactInfo>
  readonly remote?: ReadonlyArray<ContactInfo>
}

export enum NotificationType {
  InviteReceivedNotification = 'INVITE_RECEIVED',
  AccountPeerJoinedNotification = 'ACCOUNT_PEER_JOINED',
  PeerJoinedNotification = 'PEER_JOINED',
  PeerLeftNotification = 'PEER_LEFT',
  MessageAddedNotification = 'MESSAGE_ADDED',
  FilesAddedNotification = 'FILES_ADDED',
  CommentAddedNotification = 'COMMENT_ADDED',
  LikeAddedNotification = 'LIKE_ADDED'
}

export interface NotificationInfo {
  readonly id: string
  readonly date: string
  readonly actor_id: string
  readonly username?: string
  readonly avatar?: string
  readonly subject: string
  readonly subject_id: string
  readonly block_id?: string
  readonly target?: string
  readonly type: NotificationType
  readonly body: string
  readonly read: boolean
}

export interface Summary {
  readonly account_peer_cnt: number
  readonly thread_cnt: number
  readonly file_cnt: number
  readonly contact_cnt: number
}

export interface FileData {
  readonly url: string
}

export interface Link {
  readonly use?: string
  readonly pin: boolean
  readonly plaintext: boolean
  readonly mill?: string
  readonly opts?: { readonly [key: string]: string }
  readonly json_schema?: { readonly [key: string]: any }
}

export interface Node {
  readonly name?: string
  readonly pin: boolean
  readonly plaintext: boolean
  readonly mill?: string
  readonly opts?: { readonly [key: string]: string }
  readonly json_schema?: { readonly [key: string]: any }
  readonly links?: { readonly [key: string]: Link }
}

export enum ThreadType {
  PRIVATE = 'PRIVATE',
  READ_ONLY = 'READ_ONLY',
  PUBLIC = 'PUBLIC',
  OPEN = 'OPEN',
  INVALID = 'INVALID'
}

export enum ThreadSharing {
  NOT_SHARED = 'NOT_SHARED',
  INVITE_ONLY = 'INVITE_ONLY',
  SHARED = 'SHARED',
  INVALID = 'INVALID'
}

export enum ThreadState {
  LOADING = 'LOADING',
  LOADED = 'LOADED',
  INVALID = 'INVALID'
}

export interface ThreadInfo {
  readonly id: string
  readonly key: string
  readonly name: string
  readonly schema?: Node
  readonly schema_id?: string
  readonly initiator: string
  readonly type: ThreadType
  readonly sharing: ThreadSharing
  readonly members?: ReadonlyArray<string>
  readonly state: ThreadState
  readonly head?: BlockInfo
  readonly peer_cnt: number
  readonly block_cnt: number
  readonly file_cnt: number
}

export interface WalletAccount {
  readonly seed: string
  readonly address: string
}

export interface ThreadUpdate {
  block: BlockInfo
  thread_id: string
  thread_key: string
  thread_name: string
  info?: any
}

export enum UpdateType {
  ThreadAdded,
  ThreadRemoved,
  AccountPeerAdded,
  AccountPeerRemoved
}

export interface Update {
  id: string
  key: string
  name: string
  type: UpdateType
}

export enum LogLevel {
  CRITICAL = 'CRITICAL',
  ERROR = 'ERROR',
  WARNING = 'WARNING',
  NOTICE = 'NOTICE',
  INFO = 'INFO',
  DEBUG = 'DEBUG'
}
