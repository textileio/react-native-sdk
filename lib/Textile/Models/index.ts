import { AppStateStatus } from 'react-native'

export interface TextileConfig {
  RELEASE_TYPE?: string
  TEXTILE_CAFE_GATEWAY_URL?: string
  TEXTILE_CAFE_OVERRIDE?: string
  SELF_MANAGE_APP_STATE?: boolean
}

export enum NodeState {
  'nonexistent' = 'nonexistent',
  'creating' = 'creating',
  'created' = 'created', // Node has been created, on it's way to starting
  'starting' = 'starting',
  'started' = 'started',
  'stopping' = 'stopping',
  'stopped' = 'stopped', // Node has been explicitly stopped, different than created
  'creatingWallet' = 'creatingWallet',
  'derivingAccount' = 'derivingAccount',
  'initializingRepo' = 'initializingRepo',
  'walletInitSuccess' = 'walletInitSuccess',
  'postMigration' = 'postMigration'
}

export interface DiscoveredCafe {
  readonly peer: string
  readonly address: string
  readonly api: string
  readonly protocol: string
  readonly node: string
  readonly url: string
}
export interface DiscoveredCafes {
  readonly primary: DiscoveredCafe
  readonly secondary: DiscoveredCafe
}

export interface TextileOptions {
  debug?: boolean
}
export interface StoredNodeState {
  state: NodeState
  error?: string
}
export type TextileAppStateStatus = AppStateStatus | 'unknown' | 'backgroundFromForeground'

export interface File {
  readonly mill: string
  readonly checksum: string
  readonly source: string
  readonly opts?: string
  readonly hash: string
  readonly key?: string
  readonly media: string
  readonly name?: string
  readonly size: number
  readonly added: string
  readonly meta?: { readonly [key: string]: any }
  readonly targets?: ReadonlyArray<string>
}

export interface Directory {
  readonly [key: string]: File
}

export interface ThreadFileInfo {
  readonly index: number
  readonly file?: File
  readonly links?: Directory
}

export interface ThreadCommentInfo {
  readonly id: string
  readonly date: string
  readonly author_id: string
  readonly username?: string
  readonly avatar?: string
  readonly body: string
}

export interface ThreadLikeInfo {
  readonly id: string
  readonly date: string
  readonly author_id: string
  readonly username?: string
  readonly avatar?: string
}

export enum ThreadFeedItemType {
  join = 'join',
  leave = 'leave',
  files = 'files',
  message = 'message'
}

export interface ThreadFeedItem {
  readonly block: string
  readonly type: ThreadFeedItemType
  readonly join?: ThreadJoinInfo
  readonly leave?: ThreadLeaveInfo
  readonly files?: ThreadFilesInfo
  readonly message?: ThreadMessageInfo
}

export interface ThreadFilesInfo {
  readonly block: string
  readonly target: string
  readonly date: string
  readonly author_id: string
  readonly username?: string
  readonly avatar?: string
  readonly caption?: string
  readonly files: ReadonlyArray<ThreadFileInfo>
  readonly comments: ReadonlyArray<ThreadCommentInfo>
  readonly likes: ReadonlyArray<ThreadLikeInfo>
  readonly threads: ReadonlyArray<string>
}

export interface ThreadJoinInfo {
  readonly block: string
  readonly date: string
  readonly author_id: string
  readonly username?: string
  readonly avatar?: string
  readonly body: string
  readonly likes: ReadonlyArray<ThreadLikeInfo>
}

export interface ThreadLeaveInfo {
  readonly block: string
  readonly date: string
  readonly author_id: string
  readonly username?: string
  readonly avatar?: string
  readonly likes: ReadonlyArray<ThreadLikeInfo>
}

export interface ThreadMessageInfo {
  readonly block: string
  readonly date: string
  readonly author_id: string
  readonly username?: string
  readonly avatar?: string
  readonly body: string
  readonly comments: ReadonlyArray<ThreadCommentInfo>
  readonly likes: ReadonlyArray<ThreadLikeInfo>
}

export enum BlockType {
  MERGE = 'MERGE',
  IGNORE = 'IGNORE',
  FLAG = 'FLAG',
  JOIN = 'JOIN',
  ANNOUNCE = 'ANNOUNCE',
  LEAVE = 'LEAVE',
  MESSAGE = 'MESSAGE',
  FILES = 'FILES',
  COMMENT = 'COMMENT',
  LIKE = 'LIKE',
  INVALID = 'INVALID'
}

export interface BlockInfo {
  readonly id: string
  readonly thread_id: string
  readonly author_id: string
  readonly username: string
  readonly avatar?: string
  readonly type: BlockType
  readonly date: string
  readonly parents: ReadonlyArray<string>
  readonly target?: string
  readonly body?: string
}

export interface ExternalInvite {
  readonly id: string
  readonly key: string
  readonly inviter: string
}

export interface Cafe {
  readonly peer: string
  readonly address: string
  readonly api: string
  readonly protocol: string
  readonly node: string
  readonly url: string
  readonly swarm: ReadonlyArray<string>
}

export interface CafeSession {
  readonly id: string
  readonly access: string
  readonly refresh: string
  readonly exp: string
  readonly cafe: Cafe
}

export interface ContactInfo {
  readonly id: string
  readonly address: string
  readonly username?: string
  readonly avatar?: string
  readonly inboxes?: ReadonlyArray<Cafe>
  readonly created: string
  readonly updated: string
  readonly thread_ids?: ReadonlyArray<string>
}

export interface ContactInfoQueryResult {
  readonly local?: ReadonlyArray<ContactInfo>
  readonly remote?: ReadonlyArray<ContactInfo>
}

export enum NotificationType {
  InviteReceivedNotification = 'INVITE_RECEIVED',
  AccountPeerJoinedNotification = 'ACCOUNT_PEER_JOINED',
  PeerJoinedNotification = 'PEER_JOINED',
  PeerLeftNotification = 'PEER_LEFT',
  MessageAddedNotification = 'MESSAGE_ADDED',
  FilesAddedNotification = 'FILES_ADDED',
  CommentAddedNotification = 'COMMENT_ADDED',
  LikeAddedNotification = 'LIKE_ADDED'
}

export interface NotificationInfo {
  readonly id: string
  readonly date: string
  readonly actor_id: string
  readonly username?: string
  readonly avatar?: string
  readonly subject: string
  readonly subject_id: string
  readonly block_id?: string
  readonly target?: string
  readonly type: NotificationType
  readonly body: string
  readonly read: boolean
}

export interface Overview {
  readonly account_peer_cnt: number
  readonly thread_cnt: number
  readonly file_cnt: number
  readonly contact_cnt: number
}

export interface FileData {
  readonly url: string
}

export interface Link {
  readonly use?: string
  readonly pin: boolean
  readonly plaintext: boolean
  readonly mill?: string
  readonly opts?: { readonly [key: string]: string }
  readonly json_schema?: { readonly [key: string]: any }
}

export interface Node {
  readonly name?: string
  readonly pin: boolean
  readonly plaintext: boolean
  readonly mill?: string
  readonly opts?: { readonly [key: string]: string }
  readonly json_schema?: { readonly [key: string]: any }
  readonly links?: { readonly [key: string]: Link }
}

export enum ThreadType {
  PRIVATE = 'PRIVATE',
  READ_ONLY = 'READ_ONLY',
  PUBLIC = 'PUBLIC',
  OPEN = 'OPEN',
  INVALID = 'INVALID'
}

export enum ThreadSharing {
  NOT_SHARED = 'NOT_SHARED',
  INVITE_ONLY = 'INVITE_ONLY',
  SHARED = 'SHARED',
  INVALID = 'INVALID'
}

export enum ThreadState {
  LOADING = 'LOADING',
  LOADED = 'LOADED',
  INVALID = 'INVALID'
}

export interface ThreadInfo {
  readonly id: string
  readonly key: string
  readonly name: string
  readonly schema?: Node
  readonly schema_id?: string
  readonly initiator: string
  readonly type: ThreadType
  readonly sharing: ThreadSharing
  readonly members?: ReadonlyArray<string>
  readonly state: ThreadState
  readonly head?: BlockInfo
  readonly peer_cnt: number
  readonly block_cnt: number
  readonly file_cnt: number
}

export interface WalletAccount {
  readonly seed: string
  readonly address: string
}

export interface ThreadUpdate {
  block: BlockInfo
  thread_id: string
  thread_key: string
  thread_name: string
  info?: any
}

export enum UpdateType {
  ThreadAdded,
  ThreadRemoved,
  AccountPeerAdded,
  AccountPeerRemoved
}

export interface Update {
  id: string
  key: string
  name: string
  type: UpdateType
}

export enum LogLevel {
  CRITICAL = 'CRITICAL',
  ERROR = 'ERROR',
  WARNING = 'WARNING',
  NOTICE = 'NOTICE',
  INFO = 'INFO',
  DEBUG = 'DEBUG'
}
