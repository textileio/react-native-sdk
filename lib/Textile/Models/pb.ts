import * as pb from '@textile/go-mobile'
export { pb }
