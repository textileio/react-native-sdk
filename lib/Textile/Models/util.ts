import { Timestamp } from 'google-protobuf/google/protobuf/timestamp_pb'
import { Struct, Value, ListValue } from 'google-protobuf/google/protobuf/struct_pb'

import { pb } from './'

export function convertContactQueryData(data: pb.ContactQuery.AsObject) {
  const query = new pb.ContactQuery()
  query.setId(data.id)
  query.setAddress(data.address)
  query.setUsername(data.username)
  return query
}

export function convertContactQueryOptionsData(data: pb.QueryOptions.AsObject) {
  const options = new pb.QueryOptions()
  options.setExcludeList(data.excludeList)
  options.setFilter(data.filter)
  options.setLimit(data.limit)
  options.setLocal(data.local)
  options.setWait(data.wait)
  return options
}

export function convertDirectoryData(data: pb.Directory.AsObject) {
  const directory = new pb.Directory()
  for (const fileIndexData of data.filesMap) {
    directory.getFilesMap().set(fileIndexData[0], convertFileIndexData(fileIndexData[1]))
  }
  return directory
}

export function convertFileIndexData(data: pb.FileIndex.AsObject) {
  const fileIndex = new pb.FileIndex()
  fileIndex.setAdded(data.added ? convertTimestampData(data.added) : undefined)
  fileIndex.setChecksum(data.checksum)
  fileIndex.setHash(data.hash)
  fileIndex.setKey(data.key)
  fileIndex.setMedia(data.media)
  fileIndex.setMeta(data.meta ? convertStructData(data.meta) : undefined)
  fileIndex.setMill(data.mill)
  fileIndex.setName(data.name)
  fileIndex.setOpts(data.opts)
  fileIndex.setSize(data.size)
  fileIndex.setSource(data.source)
  fileIndex.setTargetsList(data.targetsList)
  return fileIndex
}

export function convertTimestampData(data: Timestamp.AsObject) {
  const timesamp = new Timestamp()
  timesamp.setSeconds(data.seconds)
  timesamp.setNanos(data.nanos)
  return timesamp
}

export function convertStructData(data: Struct.AsObject) {
  const struct = new Struct()
  for (const item of data.fieldsMap) {
    struct.getFieldsMap().set(item[0], convertValueData(item[1]))
  }
  return struct
}

export function convertValueData(data: Value.AsObject) {
  const value = new Value()
  value.setBoolValue(data.boolValue)
  value.setListValue(data.listValue ? convertListValueData(data.listValue) : undefined)
  value.setNullValue(data.nullValue)
  value.setNumberValue(data.numberValue)
  value.setStringValue(data.stringValue)
  value.setStructValue(data.structValue ? convertStructData(data.structValue) : undefined)
  return value
}

export function convertListValueData(data: ListValue.AsObject) {
  const listValue = new ListValue()
  const valuesList = data.valuesList.map((valueData) => convertValueData(valueData))
  listValue.setValuesList(valuesList)
  return listValue
}
