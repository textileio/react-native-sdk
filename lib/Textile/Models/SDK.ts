import { AppStateStatus } from 'react-native'
import { pb } from './pb'

export interface ContactSearchResult {
  contact: pb.IContact,
  local: boolean
}

export interface BufferJSON {
  buffer: string
}

export interface CafeConfig {
  TEXTILE_CAFE_GATEWAY_URL: string
  TEXTILE_CAFE_TOKEN: string
  TEXTILE_CAFE_OVERRIDE?: string
}

export enum NodeState {
  'nonexistent' = 'nonexistent',
  'creating' = 'creating',
  'created' = 'created', // Node has been created, on it's way to starting
  'starting' = 'starting',
  'started' = 'started',
  'stopping' = 'stopping',
  'stopped' = 'stopped', // Node has been explicitly stopped, different than created
  'creatingWallet' = 'creatingWallet',
  'derivingAccount' = 'derivingAccount',
  'initializingRepo' = 'initializingRepo',
  'walletInitSuccess' = 'walletInitSuccess',
  'postMigration' = 'postMigration'
}

export enum SchemaType {
  MEDIA = 'MEDIA',
  CAMERA_ROLL = 'CAMERA_ROLL',
  JSON = 'JSON'
}
export interface StoredNodeState {
  state: NodeState
  error?: string
}

export type TextileAppStateStatus = AppStateStatus | 'unknown' | 'backgroundFromForeground'
export interface TextileConfig {
  RELEASE_TYPE?: string
  SELF_MANAGE_APP_STATE?: boolean
}

export interface TextileOptions {
  debug?: boolean
}
