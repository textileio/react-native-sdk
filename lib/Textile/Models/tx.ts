import { pb } from './pb'

export type Cafe = pb.Cafe.AsObject
export type CafeChallenge = pb.CafeChallenge.AsObject
export type CafeCheckMessage = pb.CafeCheckMessages.AsObject
export type CafeClientThread = pb.CafeClientThread.AsObject
export type CafeContactQuery = pb.CafeContactQuery.AsObject
export type CafeContactQueryResult = pb.CafeContactQueryResult.AsObject
export type CafeDeleteMessages = pb.CafeDeleteMessages.AsObject
export type CafeDeleteMessagesAck = pb.CafeDeleteMessagesAck.AsObject
export type CafeDeliverMessage = pb.CafeDeliverMessage.AsObject
export type CafeMessage = pb.CafeMessage.AsObject
export type CafeMessages = pb.CafeMessages.AsObject
export type CafeNonce = pb.CafeNonce.AsObject
export type CafeObject = pb.CafeObject.AsObject
export type CafeObjectList = pb.CafeObjectList.AsObject
export type CafePubSubContactQuery = pb.CafePubSubContactQuery.AsObject
export type CafePubSubContactQueryResult = pb.CafePubSubContactQueryResult.AsObject
export type CafePublishContact = pb.CafePublishContact.AsObject
export type CafePublishContactAck = pb.CafePublishContactAck.AsObject
export type CafeRefreshSession = pb.CafeRefreshSession.AsObject
export type CafeRegistration = pb.CafeRegistration.AsObject
export type CafeSession = pb.CafeSession.AsObject
export type CafeSessions = pb.CafeSessions.AsObject
export type CafeStore = pb.CafeStore.AsObject
export type CafeStoreThread = pb.CafeStoreThread.AsObject
export type CafeStored = pb.CafeStored.AsObject
export type Comment = pb.Comment.AsObject
export type CommentList = pb.CommentList.AsObject
export type Contact = pb.Contact.AsObject
export type ContactQuery = pb.ContactQuery.AsObject
export type Directory = pb.Directory.AsObject
export type Envelope = pb.Envelope.AsObject
export type Error = pb.Error.AsObject
export type FeedItem = pb.FeedItem.AsObject
export type FeedItemList = pb.FeedItemList.AsObject
export type FeedMode = pb.FeedMode
export type File = pb.File.AsObject
export type FileIndex = pb.FileIndex.AsObject
export type Files = pb.Files.AsObject
export type FilesList = pb.FilesList.AsObject
export type Join = pb.Join.AsObject
export type JoinList = pb.JoinList.AsObject
export type Leave = pb.Leave.AsObject
export type LeaveList = pb.LeaveList.AsObject
export type Like = pb.Like.AsObject
export type LikeList = pb.LikeList.AsObject
export type Message = pb.Message.AsObject
export type MobilePreparedFiles = pb.MobilePreparedFiles.AsObject
export type PubSubQuery = pb.PubSubQuery.AsObject
export type PubSubQueryResults = pb.PubSubQueryResults.AsObject
export type Query = pb.Query.AsObject
export type QueryEvent = pb.QueryEvent.AsObject
export type QueryOptions = pb.QueryOptions.AsObject
export type QueryResult = pb.QueryResult.AsObject
export type QueryResults = pb.QueryResults.AsObject
export type QueryType = pb.QueryType
export type Text = pb.Text.AsObject
export type TextList = pb.TextList.AsObject
export type Thread = pb.Thread.AsObject
export type ThreadAnnounce = pb.ThreadAnnounce.AsObject
export type ThreadBackupQuery = pb.ThreadBackupQuery.AsObject
export type ThreadBlock = pb.ThreadBlock.AsObject
export type ThreadBlockHeader = pb.ThreadBlockHeader.AsObject
export type ThreadComment = pb.ThreadComment.AsObject
export type ThreadEnvelope = pb.ThreadEnvelope.AsObject
export type ThreadFiles = pb.ThreadFiles.AsObject
export type ThreadFlag = pb.ThreadFlag.AsObject
export type ThreadIgnore = pb.ThreadIgnore.AsObject
export type ThreadInvite = pb.ThreadInvite.AsObject
export type ThreadJoin = pb.ThreadJoin.AsObject
export type ThreadLike = pb.ThreadLike.AsObject
export type ThreadMessage = pb.ThreadMessage.AsObject
