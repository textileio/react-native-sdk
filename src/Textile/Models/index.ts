import pb from '@textile/go-mobile'

export * from './SDK'
export { pb }
