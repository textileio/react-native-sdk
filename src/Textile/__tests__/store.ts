import TextileStore from '../store'
import { NodeState, StoredNodeState } from '../Models'

const store = new TextileStore()

const storedNodeState: StoredNodeState = {
  state: NodeState.nonexistent
}

describe('rn store', () => {
  describe('defaults', () => {
    it('serialize in and out to same', async () => {
      const serializedNodeState = store.serialize(storedNodeState) as string
      expect(JSON.parse(serializedNodeState) as StoredNodeState).toEqual(storedNodeState)
    })

    it('empty state', async () => {
      expect(store.getAppState()).resolves.toMatchSnapshot()
      expect(store.getNodeOnline()).resolves.toMatchSnapshot()
      expect(store.getNodeState()).resolves.toMatchSnapshot()
    })
  })
  describe('type returns', () => {
    it('appState', async () => {
      await store.setAppState('background')
      const storedValue = await store.getAppState()
      const storedType = typeof storedValue
      expect(storedType).toMatchSnapshot()
      expect(storedValue).toMatchSnapshot()
    })
    it('nodeOnline', async () => {
      await store.setNodeOnline(true)
      const storedValue = await store.getNodeOnline()
      const storedType = typeof storedValue
      expect(storedType).toMatchSnapshot()
      expect(storedValue).toMatchSnapshot()
    })
    it('nodeOnline', async () => {
      const newState = { state: NodeState.creating }
      await store.setNodeState(newState)
      const storedValue = await store.getNodeState()
      const storedType = typeof storedValue
      expect(storedType).toMatchSnapshot()
      expect(storedValue).toMatchSnapshot()
    })
  })
})
