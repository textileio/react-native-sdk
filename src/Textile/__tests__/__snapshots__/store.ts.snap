// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rn store defaults empty state 1`] = `undefined`;

exports[`rn store defaults empty state 2`] = `undefined`;

exports[`rn store defaults empty state 3`] = `undefined`;

exports[`rn store type returns appState 1`] = `"string"`;

exports[`rn store type returns appState 2`] = `"background"`;

exports[`rn store type returns nodeOnline 1`] = `"boolean"`;

exports[`rn store type returns nodeOnline 2`] = `true`;

exports[`rn store type returns nodeOnline 3`] = `"object"`;

exports[`rn store type returns nodeOnline 4`] = `
Object {
  "state": "creating",
}
`;
